packages/flutter/test/widgets/tap_region_test.dart

```diff
@@ -1053,4 +1053,97 @@ void main() {
     await click(find.text('Outside Surface'));
     expect(tappedInside, isEmpty);
   });
+
+  testWidgets('TapRegion should register and unregister correctly on push and pop', (WidgetTester tester) async {
+    const ValueKey<String> tapRegion1Key = ValueKey<String>('TapRegion');
+    const ValueKey<String> tapRegion2Key = ValueKey<String>('TapRegion2');
+
+    int count1 = 0;
+    int count2 = 0;
+
+    final TapRegion tapRegion1 = TapRegion(
+      key: tapRegion1Key,
+      onTapOutside: (PointerEvent event) {
+        count1++;
```

Comment:
Another tiny nit based on the [style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#prefer--over-):

```suggestion
        count1 += 1;
```

(same applies to tapRegion2)